docs/IMPLEMENTATION.md

```diff
@@ -0,0 +1,207 @@
+
+**Directory Structure Flexibility:** The folder and file names in the project structure can be renamed to match your project's conventions without impacting functionality. For example, you could use `themes` instead of `styles`, `components` instead of `layers`, or `output` instead of `build`. Just update the corresponding command parameters accordingly within your project directory, and note the differences when consulting further in this documentation.
+
+### Step 3: Review Generated Structure
+
+After conversion, you'll find:
+
+```
+templates/
+├── layers/              # Individual layer definitions
+│   ├── roads.js
+│   ├── buildings.js
+│   └── labels.js
+└── styles/             # Style variants
+	├── main-style.js
+	└── alternative-style.js
+```
+
+**Key points:**
+- Each layer becomes a separate JavaScript file in `templates/layers/`
+- Each style variant gets a file in `templates/styles/`
+- The `context` section contains variables that can eliminate repetitive overrides
+- Look for opportunities to create shared variables for colors, line widths, dash patterns, zoom-based interpolations, opacity values, font sizes, and data sources
+
+### Step 4: Organize Variables (Optional)
+
+After conversion, you can organize variables to improve maintainability. This step is optional but recommended for complex styles. See the [Defining Style Variables](#defining-style-variables) section in Building From Scratch for explanations of how to organize variables.
+
+### Step 5: Test your build system
+
+```bash
+yarn map-gl-style-build \
+  --style-dir=templates/styles \
+  --layer-dir=templates/layers \
+  --out-dir=build
+```
+
+Compare the generated stylesheets in `build/` with your original files to ensure they match.
+
+### Troubleshooting Conversion
+
+**Common issues:**
+- **Missing layers**: Check that all layers from your original styles are present in the templates
+- **Incorrect styling**: Verify that layer IDs, types, and sources match between variants
+- **Missing sources**: Ensure all data sources are defined in your style templates
+
+## Building From Scratch
+
+If you're starting with no existing stylesheets, this section explains how to create the template structure manually.
+
+### Project Structure
+
+The build system organizes layer definitions as individual template files, making them easier to edit directly, track in source control, or use with external style editors. You can also define variables and give them different values by creating style variants.
+
+A simple build command exports a stylesheet for each variant, which you can reference in your Maplibre or Mapbox map.
+
+### Layer Templates
+
+The foundation of this system is the layer template - a JavaScript module that defines how a single map layer should be styled. Each layer template exports a function that receives a `context` object containing variables and returns two objects: `baseStyle` and `overrides`.
```

Comment:
Maybe worth repeating `templates/layers` here since you mention `templates/styles` below again?